### DISCUSSION TITLE
Welcome to moodle-local_deepler Discussions!

### DISCUSSION THREAD

**Discussion by brunobaudry** [brunobaudry/moodle-local_deepler#20] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by omi1kabobi** [comment]
HI Bruno, 

I am having an issue adding a glossary. 
When I try to upload my glossary (tried both .csv and .xls), I get the following error "Exception - Class "core\output\html_writer" not found"

Any suggestions?

Thanks!

Omar

**Comment by brunobaudry** [comment]
Hello Omar,

core\output\html_writer is in Moodle core since Moodle 2.0 
Which versions are you using ? (Moodle, php)

**Reply by omi1kabobi** [reply]
Thanks for the reply Bruno,

I am running: Moodle: 4.3.2 / php 8.2.29

I edited the "glossary_renderer.php" file...
On line 19,  I changed :
use core\output\html_writer;    
to:  
use html_writer;

I was then able to upload a .csv glossary and save a new glossary.

But I can't see the glossaries I already created at DeepL.

**Reply by omi1kabobi** [reply]
I changed the my default editor from Atto to TinyMCE and I can now see new glossary in DeepL site.
Changing the default editor also resolved issues to pages not being translated.

**Reply by brunobaudry** [reply]
Hi Omar thank you for the feedback. Looks like we've been light with regression testing for version under 4.5; We'll put some fixes soon...

**Reply by omi1kabobi** [reply]
Hi Bruno,
I got to say that Deepler is an awesome Moodle plugin for translations.
Great job!
Looking forward to seeing future updates.
Best regards,
Omar

**Comment by doug307** [comment]
I have experimented with DeepLer Translation plugin for Moodle over the past year or so. We wanted to know if it would work for our organization. The tests were positive. So we moved out of a test environment and now we have a persistent error. The plugin finds many sections of translatable text in our test course, however there are many sections that show no translation field. It seems that RTF fields such as description and message fields are skipped whereas the name and text fields are populated. The fields that are skipped are typically the larger fields. Can you help me get the plugin to work for all the intended fields/sections? I am attaching a screenshot to illustrate the issue.
<img width="1723" height="694" alt="DeeplerTranslation Plugin Bug" src="https://github.com/user-attachments/assets/908973e7-03b4-425c-8990-eb802328623e" />

**Reply by doug307** [reply]
Bruno, the answer to my problem with the plugin is the same as the answer to Omar's issue, at least part of it. When I forced the Tiny MCE editor in my Preferences, the missing parts of the translation fields became active. The issue seems to be resolved. I'm posting this for future travelers to see a first diagnostic step is to confirm you are using Tiny MCE for your editor.

**Reply by brunobaudry** [reply]
Thanks for the feedback Doug. What editor were you using ? So that i check if can can improve that bit
